include/envoy/http/filter.h

```diff
@@ -28,9 +28,14 @@ namespace Http {
 enum class FilterHeadersStatus {
   // Continue filter chain iteration.
   Continue,
-  // Do not iterate to any of the remaining filters in the chain. Returning
-  // FilterDataStatus::Continue from decodeData()/encodeData() or calling
+  // Do not iterate for headers on any of the remaining filters in the chain.
+  //
+  // Returning FilterDataStatus::Continue from decodeData()/encodeData() or calling
   // continueDecoding()/continueEncoding() MUST be called if continued filter iteration is desired.
+  //
+  // Note that if a local reply was sent, no further iteration for headers as well as data and
+  // trailers for the current filter and the filters following will happen. A local reply can be
+  // triggered via sendLocalReply() or encodeHeaders().
```

Comment:
Do we want similar clarification for FilterDataStatus as well?

Reply:
Doesn't hurt -- let me add something.

Reply:
added something to `FilterDataStatus::StopIterationNoBuffer`, which is what makes more sense to return after a local reply from a data callback.